moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -504,4 +504,76 @@ bool RobotTrajectory::getStateAtDurationFromStart(const double request_duration,
   return true;
 }
 
+void RobotTrajectory::print(std::ostream& out, std::vector<int> variable_indexes) const
+{
+  size_t num_points = getWayPointCount();
+  if (num_points == 0)
+  {
+    out << "Empty trajectory.";
+    return;
+  }
+
+  std::ios::fmtflags old_settings = out.flags();
+  int old_precision = out.precision();
+  out << std::fixed << std::setprecision(3);
+
+  out << "Trajectory has " << num_points << " points over " << getDuration() << " seconds\n";
+
+  if (variable_indexes.empty())
+  {
+    if (group_)
+    {
+      variable_indexes = group_->getVariableIndexList();
+    }
+    else
+    {
+      size_t variable_count = robot_model_->getVariableCount();
+      variable_indexes.reserve(variable_count);
+      for (size_t i = 0; i < variable_count; i++)
+      {
+        variable_indexes.push_back(i);
+      }
+    }
```

Comment:
```suggestion
    // use all variables
    variable_indexes.resize(robot_model_->getVariableCount());
    std::iota(variable_indexes.begin(), variable_indexes.end(), 0);
```

together with `#include <numeric>` at the top :)